pyvista/core/utilities/fileio.py

```diff
@@ -428,6 +443,28 @@ def read_plot3d(filename, q_filenames=(), auto_detect=True, attrs=None, progress
     return reader.read()
 
 
+def is_meshio_mesh(obj):
+    """Test if passed object is instance of ``meshio.Mesh``.
+
+    Parameters
+    ----------
+    obj
+        Any object.
+
+    Returns
+    -------
+    bool
+        ``True`` if ``obj`` is an ``meshio.Mesh``.
+
+    """
+    try:
+        import meshio
+
+        return isinstance(obj, meshio.Mesh)
+    except ImportError:
+        return False
```

Comment:
I don't think this would ever yield a false negative... If `meshio` isn't installed, then the passed object couldn't be a `meshio.Mesh`.

This API isn't intended to be used publically AFAIK. This is mostly an internal helper and I think this should be left as is for this PR due to how it will require code cleanup/usage elsewhere in the internal code

Reply:
This function is only used in one place in my search, in `pyvista.core.utilities.helpers` inside the `wrap` function.  It is used to check if we can wrap the object from meshio.  In this case, if we cannot import `meshio` or it is not a `meshio.Mesh` then it returns `False` and it doesn't wrap.  If we raise, we would just change the code to catch the error in `wrap` and not wrap.  This all smells like a private function to me.  This function might be better named `can_wrap_meshio` or something like this.   Or maybe the description or Returns description could be updated to reflect this nuance.  I'd vote for private usage if it is on-topic for this PR.

It might be better located directly in `pyvista.core.utilities.helpers`, or just inlined inside `wrap` and completely removing it.  This function isn't necessarily directly tied to IO, even though it is likely often (or always?) used that way.  I also do not see the value, other than avoiding a breaking change, in keeping this imported in `pyvista.core.utilities`.

But in the effort to keep this PR somewhat manageable, I think the most actionable decision is where to put it for now and whether to import it into `pyvista.core.utilities`.

Reply:
> This API isn't intended to be used publically AFAIK. This is mostly an internal helper and I think this should be left as is for this PR due to how it will require code cleanup/usage elsewhere in the internal code

OK.



> This all smells like a private function to me. This function might be better named `can_wrap_meshio` or something like this. Or maybe the description or Returns description could be updated to reflect this nuance. I'd vote for private usage if it is on-topic for this PR.

Sounds good to me (as the eventual state of this helper). Thank you both.

Reply:
Opened https://github.com/pyvista/pyvista/issues/4493 to track this.